src/test/java/com/amazon/ion/impl/bin/IonManagedWriter_1_1_Test.kt

```diff
@@ -155,7 +155,7 @@ internal class IonManagedWriter_1_1_Test {
     fun `write an encoding directive with a non-empty macro table`() {
         val expected = """
             $ion_1_1
-            $ion_encoding::((macro_table (macro null () "foo")))
+            (:$ion::set_macros (:: (macro null () "foo")))
```

Comment:
In these cases, the `$ion::` prefix is not required, right, since the IVM installs the system module? This can be out of scope for this PR, but we might as well refine this at some point to be as succinct as possible.

Reply:
Yes it's the raw writer that is determining this behavior. I'd also like to make this more succinct.

Reply:
It's the managed writer that needs to determine the behavior because the raw writer should not know whether something is in the macro table or not.

The `stepInEExp(SystemMacro)` method in the raw writer must always write a system e-expression using the specific text or binary syntax for a system e-exp (which is the current behavior).

To get rid fo the unnecessary `$ion` prefix, the managed writer need to check whether the desired system macro is in the encoding module. If it is in the encoding module, then the managed writer can use `stepInEExp(CharSequence)` or `stepInEExp(Int, Boolean, Macro)`, similar to the logic in `startMacro` (see below). If the system macro is not in the encoding module, then the managed writer needs to call `stepInEExp(SystemMacro)` (which results in the qualified invocation in text or the `EF` opcode in binary).

https://github.com/amazon-ion/ion-java/blob/576e018f55816f0a176046a0fe507b1a3eade4cf/src/main/java/com/amazon/ion/impl/bin/IonManagedWriter_1_1.kt#L845-L853